source/interact-data/modify-results.txt

```diff
@@ -0,0 +1,281 @@
+{+odm+} raises a ``Mongoid::Errors::AttributeNotLoaded`` error.
+
+You can also use the ``only()`` method to include fields from embedded
+documents. 
+
+Consider that the ``Band`` model embeds multiple ``Tour`` objects. You can
+project fields from the ``Tour`` model such as ``year``, as shown in the
+following code:
+
+.. literalinclude:: /includes/interact-data/modify-results.rb
+   :start-after: start-only-embed
+   :end-before: end-only-embed
+   :language: ruby
+   :dedent:
+
+Then, you can access the embedded fields from the returned documents:
+
+.. code-block:: ruby
+   
+   # Returns the first Tour object from
+   # the first Band in the results
+   bands.first.tours.first
+
+You can pass fields of referenced associations to the ``only()`` method,
+but the projection is ignored when loading the embedded objects. {+odm+}
+loads all fields of the referenced associations.
+
+.. note::
+
+   If you are connected to a deployment running MongoDB 4.4 or later,
+   you cannot specify an association and its fields in a projection in
+   the same query.
+
+If a document has ``has_one`` or ``has_and_belongs_to_many``
+associations, you must include the fields with foreign keys in the list
+of attributes loaded when using ``only()`` for those associations to be
+loaded.
+
+In the following example, the ``Band`` and ``Manager`` models have a
+``has_and_belongs_to_many`` association:
+
+.. code-block:: ruby
+
+   class Band
+     include Mongoid::Document
+     field :name, type: String
+     has_and_belongs_to_many :managers
+   end
+
+   class Manager
+     include Mongoid::Document
+     has_and_belongs_to_many :bands
+   end
+
+The following code demonstrates how {+odm+} can load the associated
+``Manager`` objects if you include the ``manager_ids`` field:
+
+.. literalinclude:: /includes/interact-data/modify-results.rb
+   :start-after: start-only-embed-association
+   :end-before: end-only-embed-association
+   :language: ruby
+   :dedent:
+
+Exclude Fields
+~~~~~~~~~~~~~~
+
+You can explicitly exclude fields from results by using the
+``without()`` method.
+
+The following code excludes the ``year`` field from returned ``Band``
+objects:
+
+.. literalinclude:: /includes/interact-data/modify-results.rb
+   :start-after: start-without
+   :end-before: end-without
+   :language: ruby
+   :dedent:
+
+.. important:: _id Field
+
+   {+odm+} requires the ``_id`` field for various operations, so you
+   *cannot* exclude the ``_id`` field or the ``id`` alias from results.
+   If you pass ``_id`` or ``id`` to the ``without()`` method, {+odm+}
+   ignores it.
+
+.. _mongoid-data-sort:
+
+Sort Results
+------------
+
+You can sort the order that {+odm+} returns documents by using the
+``order()`` and ``order_by()`` methods.
+
+These methods accept a hash that indicates which fields to order the
+documents by, and whether to use an ascending or descending order for
+each field.
+
+You can specify the sort direction in the following ways:
```

Comment:
S: Love the code examples for each sort variation, but seems useful to give a little more context. E.g.: "The following examples show different ways to sort by the `name` and `year` fields."

It also took me a minute to understand these examples. Seems like you can choose the sort order by passing a number or a symbol or a string? I wonder if making that point explicitly would be helpful, and possibly organizing the info in a table or something. I assume users should also be picking just one of these methods and sticking with it, to keep their code consistent. Could be a good tip to add.